src/installer/corehost/cli/deps_entry.cpp

```diff
@@ -59,18 +59,19 @@ bool deps_entry_t::to_path(const pal::string_t& base, const pal::string_t& ietf_
     pal::string_t sub_path = ietf_dir;
     append_path(&sub_path, file_path.c_str());
 
-    if (look_in_bundle && bundle::info_t::is_single_file_bundle())
+    // Don't look_in_bundle while probing a servicing location, because hammer servicing takes highest priority.
```

Comment:
Maybe better explanation would be:
"Because servicing locations are never in the bundle" - it's kind of both - servicing location takes precedence over bundle AND it can never be "in the bundle".
I'm probably just getting paranoid :smile:

Reply:
I wrote this check thinking about "If someone sets `$CORE_SERVICING` to app directory". But that's very weird and should be explicitly not supported. I'll instead add an assert to check mutual-exclusion of `is_servicing` and `look_in_bundle` cases.